### ISSUE TITLE
sbt-assembly 0.14.8 causes stack overflow

### ISSUE THREAD

**Issue by koertkuipers** [sbt/sbt-assembly#347]
i build a scala/akka application with sbt 1.2.4.
the application targets scala 2.11.12.
i use sbt-assembly 0.14.8 plugin and assemblyShadeRules for guava.

when i launch my application (which consists of the assembly jar) i get a jvm crash:
```
java.lang.VerifyError: Operand stack overflow
Exception Details:
  Location:
    akka/actor/ChildRestartStats.retriesInWindowOkay(II)Z @47: i2l
  Reason:
    Exceeded max stack size.
  Current Frame:
    bci: @47
    flags: { }
    locals: { 'akka/actor/ChildRestartStats', integer, integer, integer, long, long_2nd, long, long_2nd }
    stack: { long, long_2nd, 'java/util/concurrent/TimeUnit', integer }
  Bytecode:
    0x0000000: 2ab6 0079 0460 3eb8 0096 3704 2ab6 0098
    0x0000010: 0994 9a00 0e2a 1604 b600 9a16 04a7 0007
    0x0000020: 2ab6 0098 3706 1604 1606 65b2 00a0 1c85
    0x0000030: b600 a494 9d00 0704 a700 0403 3608 1508
    0x0000040: 9900 152a 1db6 007b 1d1b a300 0704 a700
    0x0000050: 1303 a700 0f2a 04b6 007b 2a16 04b6 009a
    0x0000060: 04ac                                   
  Stackmap Table:
    append_frame(@32,Integer,Long)
    same_locals_1_stack_item_frame(@36,Long)
    append_frame(@59,Long)
    same_locals_1_stack_item_frame(@60,Integer)
    append_frame(@81,Integer)
    same_frame(@85)
    same_locals_1_stack_item_frame(@97,Integer)

	at akka.actor.dungeon.Children$class.initChild(Children.scala:147)
	at akka.actor.ActorCell.initChild(ActorCell.scala:431)
	at akka.actor.ActorCell.supervise(ActorCell.scala:677)
	at akka.actor.ActorCell.invokeAll$1(ActorCell.scala:530)
	at akka.actor.ActorCell.systemInvoke(ActorCell.scala:545)
	at akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:283)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:235)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:10
```
reverting back to sbt-assembly 0.14.7 in my build fixes the issue.

i noticed a somewhat similar issue when googling for asm 6.2, see here:
https://gitlab.ow2.org/asm/asm/issues/317836
i dont know if its related at all but when i saw sbt-assembly upgrade to asm 6.2 i decided to try to downgrade sbt-assembly to 0.14.7 which fixed the issue.

**Comment by mzuehlke** [comment]
I ran into a similar problem using sbt-assembly 0.14.8. Executing the assembled jar the application crashed with:

`Caused by: java.lang.ClassFormatError: StackMapTable format error: bad type array size in method`

downgrading to  sbt-assembly 0.14.7 fixed that issue for me, too.

**Comment by eed3si9n** [comment]
This should be fixed now since we bumped ASM to 7.0.
